framework/include/cppmicroservices/detail/BundleAbstractTracked.tpp

```diff
@@ -110,18 +112,21 @@ void BundleAbstractTracked<S,TTT,R>::Close()
 template<class S, class TTT, class R>
 void BundleAbstractTracked<S,TTT,R>::Track(S item, R related)
 {
-  std::shared_ptr<TrackedParamType> object;
+  bool isInMap = false;
```

Comment:
The tracking map should never store nullopts, since an `AddingBundle` call that returns a `std::nullopt` necessitates that the `Bundle` is not tracked. That is why `std::optional` is used to determine whether to track, but the value type in the map is just `TrackedParamType`.

`isInMap` is solving a slightly different problem here. Previously, the tracking map was Bundles/ServiceReferences -> shared pointers. `BundleAbstractTracked` was relying on the fact that a null-constructed shared pointer is falsy as a means to check whether the key was found in the map. When we change to storing value types as the value, we do not want to force the user to define `bool()` for their tracked objects. As a result, `isInMap` is used to flag whether the key was found in the map. It is unrelated to `std::nullopt` here.